test_vectors.py

```diff
@@ -43,19 +44,18 @@ def test_vectors(self):
         ]
 
         for test_vector in test_vectors:
-            self.assertEqual(
-                balloon(
-                    test_vector["password"],
-                    test_vector["salt"],
-                    test_vector["s_cost"],
-                    test_vector["t_cost"],
-                ).hex(),
-                test_vector["output"],
-            )
+            test_params = list(test_vector.values())
+            self.assertEqual(balloon(*test_params[:4]).hex(), test_vector["output"])
             self.assertEqual(
                 balloon_hash(test_vector["password"], test_vector["salt"]),
                 balloon(test_vector["password"], test_vector["salt"], 16, 20, 4).hex(),
             )
+            self.assertTrue(verify(*test_params[:4], 3, test_vector["output"]))
+            self.assertFalse(
+                verify(
+                    *test_params[:4], 3, "%064x" % (int(test_vector["output"], 16) - 1)
```

Comment:
```py
"%064x" % (int(test_vector["output"], 16) - 1
```
What is that?

Reply:
It converts hash to int, subtracts 1 and formats it back as a 64-byte hex string. Intent is to assert that verify() will return False if the provided sha256 hash does not match.

Reply:
Interesting stuff! Can we just make a 64 byte array filled with nulls instead?

Reply:
That works too. Just looked up this, https://www.quora.com/What-data-produces-a-SHA256-hash-of-all-zero-bits. So I think it should be safe.